ui/src/app/apps/apps.service.ts

```diff
@@ -109,10 +109,12 @@ export class AppsService {
     }
     params.append('force', appRegistration.force ? 'true' : 'false');
 
-    const options = new RequestOptions({ headers: headers, params: params });
+    const options = HttpUtils.getDefaultRequestOptions();
+    options.params = params;
+
     console.log(options.params);
 
-    return this.http.post(AppsService.appsUrl + '/' + appRegistration.type + '/' + appRegistration.name, {}, options)
+    return this.http.post(AppsService.appsUrl + '/' + ApplicationType[appRegistration.type] + '/' + appRegistration.name, {}, options)
```

Comment:
In TS it returns a numeric value. While TS is generally awesome - the enum support lacks compared to Java. We should revisit once we upgrade to TypeScript `2.4` as it has support for String enums - https://blogs.msdn.microsoft.com/typescript/2017/06/27/announcing-typescript-2-4/